controllers/helmchartinstall_controller.go

```diff
@@ -58,12 +78,41 @@ func (r *HelmChartInstallReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 
 	r.Log.V(0).Info("Hello World!", "instanceName", req.NamespacedName, "instance", i)
 
+	// FIXME TODO we are conflating the setup and render phases here, because originally
+	// I was writing this controller as an experiment for a controller which only supported
+	// rendering.
+	//
+	// Here's what we want in the long run, for a controller which supports both:
+	// - Setup phase: reads configuration, and sets up event handlers. An event handler has:
+	//   * A target CRD GVK (specified by the stack configuration for the behavior)
+	//   * A watcher (the controller would create this, to watch CRDs of the desired type)
+	//   * A set of hooks (specified by the stack configuration for the behavior). Each hook
+	//     takes in a set of configuration and the triggering CRD, and outputs some K8s resources
+	// - Render phase: this is what happens when a hook is triggered, and the hook runs
+	//   * Inputs:
+	//     ^ A claim, which is the object which triggered the render
+	//     ^ A behavior, which has all of the configuration for what should happen now
+	//       that the render has been triggered
+	//   * Outputs:
+	//     ^ Changes to new or existing K8s resources
+	//     ^ The result of the render (probably as the status of the claim)
+	//       + The job which was run
+	//       + The result of the job
+	//       + The logs of the job (maybe; it might be too much)
+
+	// TODO remaining functionality for the render phase:
+	// - Surface the job result somehow (for usability)
+	// - Support for deletion, unless we assume that garbage collection will do it for us
```

Comment:
So long as we only support namespaced CRDs and only namespaced resources, a GC based deletion process seems fair.

An alternative would require looping over the dependent types of the Stack filtering by ownership labels, or using annotations to track the OID, apiversion/kind, name, and namespace of each generated resource.

Namespaced only seems like the best bet for these early controllers and demos.